### PR TITLE
Turn up maven options in mvn script

### PR DESCRIPTION
**Pull request by yaooqinn**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: build/mvn [hunk 1]
```diff
@@ -22,7 +22,7 @@ _DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 # Preserve the calling directory
 _CALLING_DIR="$(pwd)"
 # Options used during compilation
-_COMPILE_JVM_OPTS="-Xms2g -Xmx2g -Xss8m -XX:ReservedCodeCacheSize=512m"
+_COMPILE_JVM_OPTS="-Xms2g -Xmx2g -Xss8m -XX:ReservedCodeCacheSize=1g -Xss128m"
```
Comment (pan3793) [inline build/mvn:25]: Do we meet stack overflow? The default stack size is 1m, 128m is too large in my view

Comment (yaooqinn) [reply build/mvn]: No evidence for s.o. 128 is from spark

Comment (yaooqinn) [reply build/mvn]: but I suspect that the flink tests cause this as the builds mostly end up with killed there

Comment (pan3793) [reply build/mvn]: I guess it's caused by heap OOM, let wait for CI results

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**